### PR TITLE
Javascript2 week3/hiren

### PR DESCRIPTION
**Pull request by hirenkumar91**
Hi Mentor here is my home work

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -59,14 +59,23 @@ <h1>
             <div id="sortbutton">
               <button id="sortBtn" onclick="sortRecipes()">AtoZ</button>
             </div>
+           <div class="linebr"></div>
+            <div id="stopWatch">
```
Comment (vdaguenet) [inline index.html:64]: Be careful with indentation here.

File: index.html [hunk 2]
```diff
@@ -59,14 +59,23 @@ <h1>
...
+              <h1>Cooking Timer</h1>
+              <input id="cookingTime" type="number" placeholder="Insert Your Time in minuits" value= "0" required>
+              <button onclick="startCookingTimer()">start</button>
```
Comment (vdaguenet) [inline index.html]: Prefer using `addEventListener` instead of `onclick` attributes

File: index.js [hunk 3]
```diff
@@ -1,42 +1,17 @@
...
-  },
-];
-
+const recipeData = 'https://raw.githubusercontent.com/hirenkumar91/mYapi/main/recipiAppdata/data.json'
+let recipeObject = [];
 let addCount = 0;
 
+async function fetchRecipes() {
+    try {
+        const response = await fetch(recipeData);
```
Comment (vdaguenet) [inline index.js]: That's the `JS3 - week2` task already. Great job!

File: index.js [hunk 4]
```diff
@@ -102,9 +132,37 @@ const closeButton = () => {
...
+  const filteredRecipes = searchKey === ''
+    ? recipeObject
+    : recipeObject.filter(recipe => recipe.title.toLowerCase().includes(searchKey));
+  
+  console.log("Filtered Recipes:", filteredRecipes);
+
+  // Display the filtered recipes
+  displayRecipes(filteredRecipes);
 
+  console.log("this");
```
Comment (vdaguenet) [inline index.js]: Don't forget to clean up your PR from debug code before submitting it.

File: index.js [hunk 5]
```diff
@@ -179,4 +223,144 @@ function sortRecipes() {
 
   // Display the sorted recipes
   displayRecipes(sortedRecipes);
-}
\ No newline at end of file
+}
+
+//Typewriter effect
```
Comment (vdaguenet) [inline index.js:241]: Nice effect!

**Review by vdaguenet — APPROVED** [review]
You have a good base for your project and you also went ahead and implemented the features of the next weeks. So I recommend you to look into splitting your `index.js` into multiple files. For example, the typewriter effect could be in its own file.

File: index.js (hunk 3 above)
Comment (hirenkumar91) [reply index.js]: thanks

File: index.js (hunk 4 above)
Comment (hirenkumar91) [reply index.js]: thank you for suggestion

File: index.js (hunk 5 above)
Comment (hirenkumar91) [reply index.js:241]: thanks

File: index.html (hunk 2 above)
Comment (hirenkumar91) [reply index.html]: thank you for review, will do that

Comment (hirenkumar91) [reply index.html]: it's done now

**Comment by hirenkumar91** [comment]
hi to understand properly about splitting files , you mean i can categories same functions and make separate file for them .
that sounds good as i experience some time i didn't put comment and i delete something and it leads to  few thinks not working , 

for now can i merge changes and do it in separate PR

**Comment by vdaguenet** [comment]
> hi to understand properly about splitting files , you mean i can categories same functions and make separate file for them . that sounds good as i experience some time i didn't put comment and i delete something and it leads to few thinks not working ,
> 
> for now can i merge changes and do it in separate PR

Exactly @hirenkumar91, it will help you to structure your code. 
Yes you can merge this PR now. My suggestion is meant to be done when you have some free time.

**Pull request merged**